app/services/doc_auth_router.rb

```diff
@@ -185,20 +189,36 @@ def self.client(warn_notifier: nil)
           sharpness_threshold: IdentityConfig.store.doc_auth_error_sharpness_threshold,
           glare_threshold: IdentityConfig.store.doc_auth_error_glare_threshold,
         ),
+        vendor_discriminator: vendor_discriminator,
       )
     when 'mock'
       DocAuthErrorTranslatorProxy.new(
         DocAuth::Mock::DocAuthMockClient.new(
           warn_notifier: warn_notifier,
         ),
+        vendor_discriminator: vendor_discriminator,
       )
     else
-      raise "#{doc_auth_vendor} is not a valid doc auth vendor"
+      raise "#{doc_auth_vendor(discriminator: vendor_discriminator)} is not a valid doc auth vendor"
     end
   end
   # rubocop:enable Layout/LineLength
 
-  def self.doc_auth_vendor
+  def self.doc_auth_vendor(discriminator: nil)
+    if IdentityConfig.store.doc_auth_vendor_randomize
+      if discriminator.blank?
+        raise StandardError.new('doc_auth_vendor called without a session_id when randomized!')
+      end
+      target_percent = IdentityConfig.store.doc_auth_vendor_randomize_percent.clamp(0, 100)
+
+      max_sha = (16**64)-1
+      user_value = Digest::SHA256.hexdigest(discriminator).to_i(16).to_f / max_sha * 100
+
+      if user_value < target_percent
```

Comment:
Good candidate for a randomization method

Reply:
Done in `498e2e1`